extract/common.py

```
@@ -7,3 +9,22 @@ def extract_simple_pdf(data, metadata):
 
 def extract_line_numbered_pdf(data, metadata):
     return text_after_line_numbers(pdfdata_to_text(data))
+
+
+def extract_pre_tag_html(data, metadata):
+    """
+    Many states that provide bill text on HTML webpages (e.g. AK, FL)
+    have the text inside <pre> tags (for preformatted text).
+    """
+
+    html_document = BeautifulSoup(data, "lxml")
+    pre_tags = html_document.find_all("pre")
+
+    # To ensure that we exit non-zero if there are multiple <pre> tags
+    # on the page, raise an exception: this means that the extraction
+    # code needs to be updated.
+    assert len(pre_tags) == 1
```

note the assert in the extraction function - I added this such that a silent failure will not occur if the page structure changes, but let me know if there is a better practice here (with respect to eventual inclusion of this code in your pipeline)